spec/serializers/search_and_compare/course_serializer_spec.rb

```diff
@@ -309,16 +351,40 @@
           it { should include(Longitude: nil) }
           it { should include(LastGeocodedUtc: '0001-01-01T00:00:00') }
           it { should include(Address: expected_address) }
+        end
+      end
+
+      context 'DescriptionSections_Mapping' do
+        subject { resource[:DescriptionSections] }
+
+        let(:expected_enrichments) do
+          { "about this training programme": about_course,
+            "interview process": interview_process,
+            "about fees": fee_details,
+            "about salary": salary_details,
+            "entry requirements": qualifications,
+            "entry requirements personal qualities": personal_qualities,
+            "entry requirements other": other_requirements,
+            "financial support": financial_support,
+            "about school placements": how_school_placements_work,
+            "about this training provider": train_with_us,
+            "about this training provider accrediting": accrediting_provider_enrichment_description,
+            "training with disabilities": train_with_disability }.freeze
+        end
+
+        it 'map the fields' do
```

Comment:
Could we create a shared_example that does the expectations in this spec below, and then apply it to each of the expected_enrichments above? And also, without a loop, just in plain rspec?

Reply:
sorted